java/arcs/core/data/expression/Builders.kt

```diff
@@ -165,15 +171,74 @@ fun <E : Expression.Scope, T> Expression.FieldExpression<E, T>.asScope() =
     this as Expression.FieldExpression<E, Expression.Scope>
 
 /** Constructs a reference to a current scope object for test purposes */
-class CurrentScope<V>(map: Map<String, V>) : MapScope<V>("<this>", map)
+class CurrentScope<V>(map: MutableMap<String, V>) : MapScope<V>("<this>", map)
```

Comment:
Will we ever have scopes that are of a single Type? In other words should we skip the `V` type argument and use `Any` explicitly? Would save you some type casting.

Reply:
I tried removing this but it led to some other problems. I'll remove in follow ons